### PR TITLE
Make the password accessory look and behave disabled when the entry is disabled

### LINKED ISSUE #3882: Disabled RadioGroup does not display status
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

In version 2.3.4, the radio group item does not display the selected status correctly when radio group is disabled.

### How to reproduce

1. SetSelected 
2. Disable radio group

### Example code

```go
package main

import (
	"fmt"

	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Hello")

	normalGroup := widget.NewRadioGroup([]string{"A", "B", "C"}, func(string) {})
	normalGroup.SetSelected("A")

	disabledGroup := widget.NewRadioGroup([]string{"A", "B", "C"}, func(string) {})
	disabledGroup.SetSelected("A")

	// disable
	disabledGroup.Disable()

	w.SetContent(container.NewVBox(
		normalGroup,
		widget.NewLabel(fmt.Sprintf("normalGroup selected: %s", normalGroup.Selected)),
		disabledGroup,
		widget.NewLabel(fmt.Sprintf("disabledGroup selected: %s", disabledGroup.Selected)),
	))

	w.ShowAndRun()
}

```

### Fyne version

2.3.4

### Go compiler version

1.19.2

### Operating system and version

macOs Monterey

### Additional Information

_No response_

### LINKED ISSUE #3908: Disabled password Entry should also disable the ActionItem (show/hide password)
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

When an Entry with .Password=true is disabled, the Action item button should also be disabled.

### How to reproduce

Disable a Password Entry

### Screenshots

_No response_

### Example code

```go
package main

func main() {
    ...

    entry := widget.NewEntry()
    entry.Password = true
    entry.Disable()

    ...
}
```

### Fyne version

develop v2.3.4-rc1.0.20230516195251-56dd0f3e0f32

### Go compiler version

1.20

### Operating system and version

Windows 10

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by andydotxyz**
Fixes #3908

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

### CODE REVIEW

**Comment by Jacalz** [comment]
You had accidentally pasted the wrong commit into the commit. I updated the PR description with #3908 but I would recommend that you squash merge the PR to be able to change the description of the commit before merging.

File: widget/entry_test.go [hunk 1]
```diff
@@ -1736,6 +1736,19 @@ func TestPasswordEntry_ActionItemSizeAndPlacement(t *testing.T) {
 	assert.Equal(t, fyne.NewPos(e.MinSize().Width-2*theme.Padding()-b.Size().Width, 2*theme.Padding()), b.Position())
 }
 
+func TestPasswordEntry_Disabled(t *testing.T) {
+	entry, window := setupPasswordTest(t)
+	defer teardownImageTest(window)
```
Comment (Jacalz) [inline widget/entry_test.go:1741]: Do you really need to do this when not using any image tests?

BTW: I noticed that that function is writing to the same markup test file on each call which seems unnecessary. Will push a new PR to fix that.

**Review by Jacalz — APPROVED** [review]
I realised that the new code is in liken worth the old code and isn't technically doing anything wrong. I'll fix the function on develop some time in the future to make sure that we aren't updating that markup on each test setup.

**Comment by andydotxyz** [comment]
Thanks for fixing it up. I am over tired this week!

**Pull request merged**